### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

**Potential roadblock** https://github.com/SciTools/iris/pull/6195

Good news! The race to support Python 3.13 is a **lot** shorter than 3.12, we have the following roadblocks, in the process of unblocking:

- ~iris needs be built with numpy>=2 since **only** numpy>=2 has support for Python 3.13; iris=3.11 is almost out, and that is built with numpy>=2 and 3.13, we need to get in https://github.com/ESMValGroup/ESMValCore/pull/2395 ourselves~

That's it! We fixed esgf-pyclient for 3.13 recently (today), and all else goes through!

~**Prerequisites** needs https://github.com/ESMValGroup/ESMValCore/pull/2395~ merged via `main` merge in d1dc683f3564b61117eb5690f24f5bc704565842

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
@ESMValGroup/technical-lead-development-team @trexfeathers @ESadek-MO some very good news wrt ESMValCore and iris=3.11 and Numpy>2 and Python 3.13 - using the release candidate for iris=3.11 we are now fully covered for Numpy>2 and Python 3.13 support - environment solves smoothly, and the tests that fail are the ones I fixed in https://github.com/ESMValGroup/ESMValCore/pull/2395 (ie all related to Numpy 2.0+ API and functionality/deprecations); am very happy since the transition to both these important dependencies' updates (Numpy and Python, no sweat, iris :grin: ) is all but complete, without having to go though the hell I went through for Python 3.12 support :partying_face: :beers:

**Comment by valeriupredoi** [comment]
iris==3.11 is out as of 10min ago https://anaconda.org/conda-forge/scitools-iris

**Comment by valeriupredoi** [comment]
look gents @bouweandela @schlunma - green :green_circle: - we good to get in Python 3.13 :partying_face:

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ on:
   push:
     branches:
       - main
+      - python313
```
Comment (bouweandela) [inline .github/workflows/run-tests.yml:24]:
```suggestion
```

**Review by bouweandela — APPROVED** [review]
Thanks V!

**Comment by valeriupredoi** [comment]
That's a good find @bouweandela - they should have pinned Python in their conda recipe if 3.13 is a no-go, will ping them myself, see if I can expedite the adoption 🍺

**Comment by bouweandela** [comment]
Maybe no code changes are needed in Iris to support Python 3.13, since the tests here are all passing fine. Let's juts keep an eye on that PR and with a bit of luck things will just work.

**Comment by valeriupredoi** [comment]
hey bud @bouweandela have a look at this https://github.com/SciTools/iris/pull/6195#issuecomment-2470676025 , that means we good to go with this :beer:

**Review by bouweandela — APPROVED** [review]

**Comment by valeriupredoi** [comment]
legend @bouweandela many thanks :beers:

**Pull request merged**